large-message-core/src/main/java/com/bakdata/kafka/LargeMessageRetrievingClient.java

```diff
@@ -71,7 +73,14 @@ public byte[] retrieveBytes(final byte[] data, final Headers headers, final bool
         }
         final LargeMessagePayloadProtocol protocol = getProtocol(headers, isKey);
         final LargeMessagePayload payload = protocol.deserialize(data, headers, isKey);
-        return this.getBytes(payload);
+        CompressionType compressionType = CompressionType.NONE;
+        Header compressionHeader = headers.lastHeader(CompressionType.headerName);
+        if (compressionHeader != null) {
+            compressionType = CompressionType.forId(compressionHeader.value()[0]);
+            headers.remove(CompressionType.headerName);
```

Comment:
Headers should only be removed in the Deserializer (https://github.com/bakdata/kafka-large-message-serde/blob/master/large-message-serde/src/main/java/com/bakdata/kafka/LargeMessageDeserializer.java#L81). For Kafka connect we need to retain the header in order to send dead letter messages